### PR TITLE
Ros2

### PR DESCRIPTION
**Pull request by griswaldbrooks**

### CODE REVIEW

File: ros2/rcl_handle/include/rcl_handle/action.hpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+};
+
+/// \brief Wrapper for action server
+/// \tparam T Action message type
+template <typename T>
+struct action_server final : public interface::action_server<T> {
+  action_server(std::shared_ptr<rclcpp::Node> node, std::string topic)
+      : node_{std::move(node)}, topic_{std::move(topic)} {}
+
+  void register_handles(goal_callback<T> goal, cancel_callback<T> cancel,
```
Comment (abake48) [inline ros2/rcl_handle/include/rcl_handle/action.hpp:148]: Just to make sure I'm following, the idea is to use the Type Erasure idiom to DI the action server callbacks, right?

**Review by abake48 — COMMENTED** [review]
Couple comments on my first pass to make sure I'm following correctly

File: ros2/rcl_handle/include/rcl_handle/action.hpp (hunk 1 above)
Comment (griswaldbrooks) [reply ros2/rcl_handle/include/rcl_handle/action.hpp:148]: That will be but there is no type erasure in this PR

File: ros2/README.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+"/usr/lib/ccache/cc"
+
+  is not able to compile a simple test program.
+```
+Likely, the host ccache directory in your [home directory](#ccache) was not created properly.
+Ensure that the directory is created and owned by the host user.
+```
+mkdir -p ~/.development-container/ccache
+sudo chown -R $(id -u):$(id -g) ~/.development-container/ccache
+```
```
[the remark this answers is not in the document]
Comment (griswaldbrooks) [reply ros2/README.md:213]: Do you want to write one? We can add it to the development container repo as well

**Pull request merged**